consensus/istanbul/validator/default.go

```diff
@@ -237,11 +283,20 @@ func (valSet *defaultSet) RemoveValidators(removedValidators *big.Int) bool {
 func (valSet *defaultSet) Copy() istanbul.ValidatorSet {
 	valSet.validatorMu.RLock()
 	defer valSet.validatorMu.RUnlock()
-	newValSet := NewSet(MapValidatorsToData(valSet.validators))
+	newValSet := NewSetFromDataWithBLSKeyCache(MapValidatorsToDataWithBLSKeyCache(valSet.validators))
```

Comment:
i'd suggest we instead have `Validator.Copy()` that will copy the uncompressed value if it has one. And so, this method doesn't really have to handle with the cache.

And then you just do:
```go
	return &defaultSet{
    validators: copiedValidators,
    randomness: randomness
  }
```

If not, what we are doing is actually transforming all validators to data, and then back again to valiators.. no sense on doing that